sycl/source/detail/scheduler/commands.cpp

```diff
@@ -943,11 +945,20 @@ cl_int ExecCGCommand::enqueueImp() {
 
     ReverseRangeDimensionsForKernel(NDRDesc);
 
-    PI_CALL(RT::piEnqueueKernelLaunch, MQueue->getHandleRef(), Kernel,
-            NDRDesc.Dims, &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
-            HasLocalSize ? &NDRDesc.LocalSize[0] : nullptr, RawEvents.size(),
-            RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
-
+    pi_result Error = PI_CALL_RESULT(
+        RT::piEnqueueKernelLaunch, MQueue->getHandleRef(), Kernel, NDRDesc.Dims,
+        &NDRDesc.GlobalOffset[0], &NDRDesc.GlobalSize[0],
+        HasLocalSize ? &NDRDesc.LocalSize[0] : nullptr, RawEvents.size(),
+        RawEvents.empty() ? nullptr : &RawEvents[0], &Event);
+
+    if (PI_SUCCESS != Error) {
+      // Analyze return code in order to convert pi_error to a corresponding
+      // exception describing a possible problem
```

Comment:
Looks like this comment should be somewhere around `detail::enqueue_kernel_launch::handleError` declaration.

Reply:
Refactored this comment and added some description to the declaration